ci/travis_script_toolchain.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License. See accompanying LICENSE file.
+
+set -xe
+
+# Use a C++11 compiler on Linux
+export CC="gcc-4.9"
+export CXX="g++-4.9"
+
+# ----------------------------------------------------------------------
+# Set up external toolchain
+
+MINICONDA_URL="https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh"
+export MINICONDA=$HOME/miniconda
+wget -O miniconda.sh $MINICONDA_URL
+bash miniconda.sh -b -p $MINICONDA
+export PATH="$MINICONDA/bin:$PATH"
+export CPP_TOOLCHAIN=$TRAVIS_BUILD_DIR/cpp-toolchain
+
+conda update -y -q conda
+conda config --set auto_update_conda false
+conda info -a
+
+conda config --set show_channel_urls True
+
+# Help with SSL timeouts to S3
+conda config --set remote_connect_timeout_secs 12
+
+conda info -a
+
+conda create -y -q -p $CPP_TOOLCHAIN \
+      boost-cpp zlib thrift-cpp snappy brotli cmake git \
```

Comment:
In pending PARQUET-981 [changelist](https://github.com/apache/parquet-cpp/pull/329) I've put [specific versions](https://github.com/apache/parquet-cpp/pull/329/files#diff-180360612c6b8c4ed830919bbb4dd459R43) of libs for "Toolchain" Appveyor configuration
`conda install -y boost-cpp=1.63 arrow-cpp=0.3.0 brotli=0.6.0 zlib=1.2.11 snappy=1.1.4 thrift-cpp=0.10.0`